src/persist/src/indexed/mod.rs

```diff
@@ -198,13 +199,29 @@ impl<K: Data, V: Data, U: Buffer, L: Blob> Indexed<K, V, U, L> {
                 entry
                     .updates
                     .iter()
-                    .map(|((key, val), ts, diff)| (seqno, (key.clone(), val.clone()), *ts, *diff)),
+                    .map(|((key, val), ts, diff)| ((key.clone(), val.clone()), *ts, *diff)),
             );
+
+            // Sanity check that each sequence number occurs exactly once.
```

Comment:
I don't think we care tbh. Is there anything that would break if they were? We really only care about the seqnos as a way to atomically swap things from buffer into future

Reply:
I guess the property we care about is "once we observe a range of sequence numbers [lower, upper) in a snapshot, we can truncate and only look at sequence numbers >= upper" which works fine with repeating sequence numbers so long the buffer's snapshot implementation is smart enough to only display sequence numbers that have been "closed"

Reply:
i removed the check